### ISSUE TITLE
Command string comparisons should be case-insensitive

### ISSUE THREAD

**Issue by jbilcke** [chadxz/imap-simple#5]
Hi,

I've encountered an issue on a real-world IMAP server, where message codes are sent back in lower case (`part.encoding` was "base64" and not "BASE64")

It would be more robust to have comparisons made case insensitive 

I think the spec also recommends that:

```
The use of upper or lower case
characters to define token strings is for editorial clarity
only.  Implementations MUST accept these strings in a
case-insensitive fashion.
```

(imap v4, but it is probably the same for other versions)
https://tools.ietf.org/html/rfc3501

**Comment by chadxz** [comment]
Hey @jbilcke , thanks for bringing this up. I'll try to get this fixed up when I have some time, but I'd also be happy to accept a PR if you're up for it.

**Comment by jbilcke** [comment]
@chadxz don't worry I already have a patch ready, I will submit the PR later today when I get back to my home

**Comment by chadxz** [comment]
Closed by #6
